src/behaviors/loading/loading.js

```diff
@@ -0,0 +1,28 @@
+(function() {
+'use strict';
+
+angular
+  .module('semantic.ui.behaviors.loading', [])
+
+  .directive('smLoading', smLoading);
+
+  smLoading.$inject = ['$parse'];
+
+  function smLoading($parse) {
+    return {
+      restrict:'A',
+      link: function(scope, element, attrs) {
+        scope.$watch(function() {
+          return $parse(attrs.smLoading)(scope);
+        }, function(loading) {
+          if (loading) {
+            element.addClass('loading');
```

Comment:
In complex applications, this directive improvement the code readability, like a smButton directive. 
Example using ngClass:  
```html
<div class="ui segment" ng-class="{'loading': ctrl.cacheChecking}">
  <h1>Posts</h1>
  <div class="ui segment" ng-class="{'loading': ctrl.objects.length === 0}">
    <button class="ui button" ng-class="{'loading: ctrl.saving'}">
      Save
    </button>
  </div>
  
  <form class="ui form" ng-class="{'loading: ctrl.saving'}">
    <!-- ... -->
    <button class="ui button">
      Save
    </button>
  </form>
</div>
```

Example using smLoading:  
```html
<div class="ui segment" sm-loading="ctrl.cacheChecking">
  <h1>Posts</h1>
  <div class="ui segment" sm-loading="ctrl.objects.length === 0">
    <button class="ui button" sm-loading"ctrl.saving">
      Save
    </button>
  </div>
  
  <form class="ui form" sm-loading="ctrl.saving">
    <!-- ... -->
    <button class="ui button">
      Save
    </button>
  </form>
</div>
```

The loading class use is very common in application using semantic-ui and ajax, like Single Page Application.